src/components/Choropleth.vue

```diff
@@ -34,43 +34,39 @@ import { APP_BASE_ROUTE } from "../router/utils";
 import { debounce } from "../utils";
 import { backgroundLayer } from "./utils";
 import { useLoadingSpinner } from "../composables/useLoadingSpinner";
+import { useSelectedMapInfo } from "../composables/useSelectedMapInfo";
 
 const mapLoading = ref(true);
 const router = useRouter();
-const { selectedFeatures, selectedIndicators, selectedIndicator, selectedCountryId, appConfig } =
-    storeToRefs(useAppStore());
+const { mapSettings, appConfig } = storeToRefs(useAppStore());
 const featureProperties = appConfig.value.geoJsonFeatureProperties;
 
 const { tooltipForFeature } = useTooltips();
-const { getFillAndOutlineColour } = useColourScale(selectedIndicators);
+const { getFillAndOutlineColour } = useColourScale();
 
 const featureInSelectedCountry = (feature: Feature, selectedCountry: string) =>
     feature.properties[featureProperties.country] === selectedCountry;
 
-const featureAdminLevel = (feature: Feature, selectedCountry) => {
-    return featureInSelectedCountry(feature, selectedCountry) &&
-        !appConfig.value.countriesWithoutAdmin2.includes(selectedCountry)
-        ? 2
-        : 1;
+const getFeatureProp = (feature: Feature, key: "name" | "id") => {
+    const isAdmin2 = featureInSelectedCountry(feature, mapSettings.value.country) && mapSettings.value.adminLevel === 2;
+    let propertyKey: string;
+    if (key === "id") {
+        propertyKey = isAdmin2 ? featureProperties.idAdm2 : featureProperties.idAdm1;
+    } else {
+        propertyKey = isAdmin2 ? featureProperties.nameAdm2 : featureProperties.nameAdm1;
+    }
+    return feature.properties[propertyKey];
```

Comment:
yh i kinda agree, it just felt like the two functions were doing the same thing, i have a better idea for how to make this function cos i realised that we control feature properties, have had another go at the function, let me know what you think, if its still less readable, happy to revert back to the old functions

Reply:
also yes we should not have to worry about `countriesWithoutAdmin2` that kind of consideration should be made in only one place i think, otherwise it is easy to forget, and we already do it in the store

Reply:
I respect the aversion to duplication, but also a little is fine now and then. https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
fair enough! have reverted it back (with a tiny change)